### ISSUE TITLE
Bug: proxy continues emitting events to disposed observer

### ISSUE THREAD

**Issue by axefrog** [mostjs-community/most-proxy#4]
I have a small bit of code in which I'm using a proxy:

``` js
function proxy(initial) {
  const {attach, stream} = makeProxy();
  return [stream.startWith(initial), attach];
}

export default function flatScan(f, initial) {
  return stream => {
    const [proxyState$, fulfill] = proxy(initial);
    const state$ = most.just(proxyState$).join(); // prevent proxy disposal bug
    const initialize = (x, state) => f(state, x, state$);
    const nextState$ = sample(initialize, stream, state$)
      .join()
      .thru(fn$ => sample((fn, state) => fn(state), fn$, state$))
      .startWith(initial)
      .multicast();
    fulfill(nextState$);
    return state$;
  };
}
```

Notice the commented line where I'm wrapping the proxy stream in another stream and then flattening it again. If I don't do this, and a second observer attaches to the proxy before the previous observer has disposed itself, then the original observer will still try to call dispose as it should, but the proxy will continue emitting events to that observer's sink, even after disposal. Somehow wrapping the proxy in another dummy operator solves the problem.

If you need an active use case in order to figure this out, let me know and I'll help you set up a working copy of my project so you can see it happening.

**Comment by TylorS** [comment]
Thank you @axefrog I will look into this ASAP

**Comment by TylorS** [comment]
@axefrog Also as a fun fact `you can use `fulfill` inside of `.thru()`

It simply returns the stream passed to it 

```js
function proxy(initial) {
  const {attach, stream} = makeProxy();
  return [stream.startWith(initial), attach];
}

export default function flatScan(f, initial) {
  return stream => {
    const [proxyState$, fulfill] = proxy(initial);
    const state$ = most.just(proxyState$).join(); // prevent proxy disposal bug
    const initialize = (x, state) => f(state, x, state$);
    const nextState$ = sample(initialize, stream, state$)
      .join()
      .thru(fn$ => sample((fn, state) => fn(state), fn$, state$))
      .startWith(initial)
      .multicast()
      .thru(fulfill);
    return state$;
  };
}
```

**Comment by axefrog** [comment]
Ah, very concise, thanks for the tip!

**Comment by TylorS** [comment]
@axefrog If you could provide me with a minimal test case, that wouhld be very helpful. I've not been able to come up with one on my own.

**Comment by TylorS** [comment]
Here's an example from @whitecolor http://www.webpackbin.com/Vyg5aOS4-

**Comment by whitecolor** [comment]
@axefrog you may try also https://github.com/whitecolor/cycle-proxy it is a like, but more general solution implemented using cycle adapters.

**Comment by TylorS** [comment]
This should be fixed with version 3.2.0.

I have a test for this now, modified from the webpackbin @whitecolor made, thanks for that again @whitecolor. That test is [here](https://github.com/mostjs-community/most-proxy/blob/master/test/index.ts#L25)
